File: Recursion/snowflake.ts

function drawSide(lineLength: number, p: Pen) {
    if (lineLength > 5) {
        const aThird = lineLength / 3;
        drawSide(aThird, p);
        p.turn(60);
        drawSide(aThird, p);
        p.turn(-120);
        drawSide(aThird, p)
        p.turn(60);
        drawSide(aThird, p)
    }
    else {
        p.go(lineLength);
    }
}

function drawFlake(size: number, p: Pen) {
   for (let n = 1; n <= 3; n++) {
        drawSide(size, p);
        p.turn(-120);
    }
}

let p = new Pen("mycanvas");
p.jump(250, 250);
drawFlake(200, p);
p.draw();

File: TurtleGraphics/Snowflake/SnowFlake.cs
using Nakov.TurtleGraphics;

namespace TurtleGraphics
{
    public static class SnowFlake
    {
        public static void Draw(int size, int level)
        {
            Turtle.PenSize = 1;
            for (int n = 1; n <= 3; n++)
            {
                DrawSide(size, level);
                Turtle.Rotate(-120);
            }
        }

        private static void DrawSide(int lineLength, int level)
        {
            if (level <= 1)
            {
                Turtle.Forward(lineLength);
            }
            else
            {
                int aThird = lineLength / 3;
                DrawSide(aThird, level - 1);
                Turtle.Rotate(60);
                DrawSide(aThird, level - 1);
                Turtle.Rotate(-120);
                DrawSide(aThird, level - 1);
                Turtle.Rotate(60);
                DrawSide(aThird, level - 1);
            }
        }
    }
}
